### DISCUSSION TITLE
Moes ZHT-3000 Serie

### DISCUSSION THREAD

**Discussion by MarkoM-71** [Koenkk/zigbee2mqtt#16009] [Devices]
Hello, I bought 3 Moes ZHT-3000 wall thermostats and integrated them into Home Assistant via zigbee2mqtt. However, they are recognized there as BHT-002-GCLZB. I can set the target temperature in 0.5°C increments on the thermostat. Unfortunately, there are only 1°C steps over z2m.
Can someone adjust this please?

**Comment by git-greatguy** [comment]
same problem with moes TS0601 _TZE200_b6wax7g0.

Trying to make custom converter from moes.js.

modifying this line for this specific model : 
from :
```
.withLocalTemperature(ea.STATE).withSetpoint('current_heating_setpoint', 5, 35, 1, ea.STATE_SET)
.withLocalTemperatureCalibration(-9, 9, 1, ea.STATE_SET)
```

to

```
.withLocalTemperature(ea.STATE).withSetpoint('current_heating_setpoint', 5, 35, 0.1, ea.STATE_SET)
.withLocalTemperatureCalibration(-9, 9, 0.1, ea.STATE_SET)

```

Interface now permit float values with 0.1 increment, but this float values seems to be cast back to integer elsewhere in z2m code even if i set a ".5" values that should be accepted by thermostat.
looking futher...

**Reply by MarkoM-71** [reply]
Where can i find that file? Oh how can i make it?

**Comment by git-greatguy** [comment]
I dig a bit into the files, so values sent related to file "toZigbee.js".
My moes TS0601 _TZE200_b6wax7g0 are defined in "moesS" declaration.

found this 
```
moesS_thermostat_current_heating_setpoint: {
        key: ['current_heating_setpoint'],
        convertSet: async (entity, key, value, meta) => {
            const temp = Math.round(value);
            await tuya.sendDataPointValue(entity, tuya.dataPoints.moesSheatingSetpoint, temp);
        },
    },
```
Tried to change Math.round(value); to value.toFixed(1) but values doesnt solve problem. Values set on thermostat goes to "00" and data read back are > 1000.

There are futher investigation to do in tuya files... i have no more time to sped for now on this. We really need a z2m dev to investigate this.

**Reply by ivan11paths** [reply]
I have a BHT-002-GCLZB thermostat and I have the same problem. From what I understood the firmware does not support 0.5 ºC, but I hoped that the new models would have solved the problem. There is a thread about this from a while ago https://github.com/Koenkk/zigbee2mqtt/discussions/6453
